src/core/midi/GOMidiEvent.h

```diff
@@ -55,6 +55,7 @@ class GOMidiEvent {
   GOTime m_time;
   wxString m_string;
   std::vector<uint8_t> m_data;
+  bool m_useNoteOff;
```

Comment:
> So GOMidiSenderEventPattern is a better place for m_useNoteOff

It's already there, inherited from GOMidiEventPattern. But the actual MIDI event still needs to be created in GOMidiEvent, itself needs to know if it is to send 0x8X, or 0x9X with velocity 0, and at that point no such information exist as it's washed away in GOMidiSender unless transferred to the actual event.

Rearranging the internal MIDI type system is outside this PR, though I agree that the whole MIDI system would need an overhaul since it's quite un-clear... Internally, GO never uses 0x8X, it's only used when sending note off so a MIDI_NOTE_OFF is kind of a contradiction. Also, as stated in #1640 the MIDI standard specify the 0x9X with velocity 0 as valid note off (which is what is used internally in GO) - and that is still in the end just a MIDI_NOTE.